// react-hexagon/lib/BackgroundDef.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _React$PropTypes = _react2.default.PropTypes;
var number = _React$PropTypes.number;
var string = _React$PropTypes.string;
var bool = _React$PropTypes.bool;


function getSize(props) {
  var width = props.diagonal;
  var height = props.diagonal;

  if (props.flatTop) {
    width *= 1.05;
    height *= 1.05;
  }

  if (props.backgroundScale) {
    width *= props.backgroundScale;
    height = width;
  } else if (props.backgroundWidth) {
    width = props.backgroundWidth;
    height = props.backgroundHeight;
  } else if (props.backgroundSize) {
    width = props.backgroundSize;
    height = props.backgroundSize;
  }

  return { width: width, height: height };
}

function BackgroundDef(props) {
  var diagonal = props.diagonal;

  var _getSize = getSize(props);

  var width = _getSize.width;
  var height = _getSize.height;


  var hasSize = width !== diagonal;
  var imgWidth = hasSize ? width : '100%';
  var imgHeight = hasSize ? height : '100%';
  var yOffset = props.yOffset || 0;
  var xOffset = typeof props.xOffset === 'undefined' && props.flatTop ? 0 - imgWidth * 0.065 : 0;

  return _react2.default.createElement(
    'defs',
    null,
    _react2.default.createElement(
      'pattern',
      { id: props.id, width: width, height: height, patternUnits: 'userSpaceOnUse' },
      _react2.default.createElement('image', { x: xOffset, y: yOffset, width: imgWidth, height: imgHeight, xlinkHref: props.backgroundImage })
    )
  );
}

BackgroundDef.propTypes = {
  id: string.isRequired,
  flatTop: bool,
  backgroundImage: string.isRequired,
  backgroundScale: number,
  backgroundWidth: number,
  backgroundHeight: number,
  backgroundSize: number,
  xOffset: number,
  yOffset: number,
  diagonal: number
};

exports.default = BackgroundDef;
